src/plugins/intel_cpu/src/nodes/reorder.cpp

```diff
@@ -376,45 +331,283 @@ void Reorder::reorderData(const IMemory &input, const IMemory &output, MultiCach
+    //                input(PrecX) --> reorder(PrecX) --> conversion(PrecX->PrecY) --> output(PrecY)
+    //            If PrecisionZ == PrecisionY, do data conversion before dnnl::reorder
+    //                input(PreX) --> conversion(PrecX->PrecY) -> reorder(PrecY) --> output(PrecY)
+    // [case 3]: Input/output precision are the different precision with same layout, in this case only precision is needed
+    //    Don't need do reorder, only do data conversion
+    //
+
+    const auto src_prc = src->getDesc().getPrecision();
+    const auto dst_prc = dst->getDesc().getPrecision();
+    OPENVINO_ASSERT(src_prc != InferenceEngine::Precision::UNSPECIFIED,
+                    "ReorderExecutor input precision is unspecified!");
+    OPENVINO_ASSERT(dst_prc != InferenceEngine::Precision::UNSPECIFIED,
+                    "ReorderExecutor output precision is unspecified!");
+    prim = dnnl::reorder();
+
+    if (src_prc != dst_prc) {
+        auto src_bmd = src->getDescWithType<BlockedMemoryDesc>();
+        auto dst_bmd = dst->getDescWithType<BlockedMemoryDesc>();
+        // dnnl::reorder data conversion is number rounding
+        // cpu_convert() data conversion is number truncated
+        // For float -> integer, still create dnnl::reorder to avoid truncation difference
+        if (src_bmd->getOrder() == dst_bmd->getOrder() && !(src_prc.is_float() && !dst_prc.is_float())) {
+            need_reorder = false;
+            src_blocked = std::make_shared<Memory>(engine, src->getDescPtr(), src->getData(), false);
+            dst_blocked = std::make_shared<Memory>(engine, dst->getDescPtr(), dst->getData(), false);
+            pre_converter = std::make_shared<IntermConverter>(src_blocked, src_prc, dst_blocked, dst_prc);
+            return;
+        }
+    }
+
+    need_reorder = true;
+    const auto src_data_type = DnnlExtensionUtils::IEPrecisionToDataType(src_prc);
+    const auto dst_data_type = DnnlExtensionUtils::IEPrecisionToDataType(dst_prc);
+
+    // dnnl::memory::data_type doesn't support this precision
+    if (src_data_type == dnnl::memory::data_type::undef && dst_data_type == dnnl::memory::data_type::undef) {
+        OPENVINO_THROW("Reorder doesn't support: ", src_prc, " -> ", dst_prc);
+    }
+
+    src_blocked = std::make_shared<Memory>(engine, src->getDescPtr(), src->getData(), false);
+    dst_blocked = std::make_shared<Memory>(engine, dst->getDescPtr(), dst->getData(), false);
+
+    dnnl::memory::desc src_desc, dst_desc;
+    if (src_data_type != dnnl::memory::data_type::undef && dst_data_type != dnnl::memory::data_type::undef) {
+        // Check whether dnnl::reorder directly support input->output
+        src_desc = updateSrcDesc(engine, src_permutation);
+        dst_desc = dst_blocked->getPrimitive().get_desc();
+        auto result = getReorderPrim(cache, engine, src_desc, dst_desc);
+
+        if (result) {
+            prim = result;
```

Comment:
I suppose this is a hot path for many cases, so can we change line 399 to avoid unnecessary construction?

Reply:
line 399 is used to construct an empty dnnl::reorder, which is an empty object and is used to check whether need do reorder (in case of only do convert).